pick_kernel/kernel.py

```diff
@@ -303,7 +350,47 @@ async def async_execute_request(self, stream, ident, parent):
                     display_id=kernel_display_id,
                     parent=parent,
                 )
-                self.child_kernel = await self.start_kernel(kernel_name, config)
+                try:
+                    self.child_kernel = await self.start_kernel(kernel_name, config)
+                except PickRegistrationException as err:
+                    # Get access to the exception info prior to doing any potential awaiting
+                    exc_info = sys.exc_info()
+                    self.log.info(exc_info)
+
+                    separator = "\n"
+
+                    self.display(
+                        Markdown(
+                            f"""## There is no kernel magic named `{kernel_name}`
+
+These are the available kernels: 
+
+{separator.join([f"* `{kernel}`" for kernel in _subkernels.list_subkernels()])}
+
+                        """
+                        ),
+                        display_id=kernel_display_id,
+                        update=True,
+                        parent=parent,
+                    )
+                    await asyncio.sleep(0.1)
```

Comment:
Why the waits?

Reply:
Good catch! I put these in while I was trying to figure out why the frontend was not resetting the `[*]` then I finally realized it was because I wasn't passing the `idents` to `session.send`.